pkg/workflows/management/core_components.go

```diff
@@ -39,8 +39,7 @@ type upgradeCoreComponents struct {
 	UpgradeChangeDiff *types.ChangeDiff
 }
 
-// Run upgradeCoreComponents upgrades pre cluster upgrade components.
-func (s *upgradeCoreComponents) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+func runUpgradeCoreComponents(ctx context.Context, commandContext *task.CommandContext) error {
```

Comment:
nit: we can drop `run` since it doesn't add any new info
```suggestion
func upgradeCoreComponents(ctx context.Context, commandContext *task.CommandContext) error {
```

Reply:
I tried it, but upgradeCoreComponents function name is conflicting with the struct name "upgradeCoreComponents"

Reply:
got you
maybe rename the task to `upgradeCoreComponentsTask`?

Reply:
for what are you optimizing by having this function at all? Is it redundant to `func (s *upgradeCoreComponents) Run`?

Reply:
sounds good to me